src/odemis/acq/align/test/fastem_test.py

```diff
@@ -0,0 +1,178 @@
+        f = fastem.align(self.scanner, self.multibeam, self.descanner, self.mppc, self.stage, self.ccd,
+                         self.beamshift, self.det_rotator, calibrations)
+
+        e = f.result(timeout=900)
+
+        self.assertIsNone(e)  # check no exceptions were returned
+        # check that z stage position is close to good position
+        # Note: This accuracy is dependent on the value chosen for the magnification on the lens.
+        numpy.testing.assert_allclose(self.focuser.position.value["z"], self.good_focus, atol=2e-6)
+
+    def test_image_translation_prealign(self):
+        """Run the image translation prealing calibration. Can also be tested with simulator.
+        It calibrates the descanner offset."""
+
+        calibrations = [IMAGE_TRANSLATION_PREALIGN]
+
+        # get current descanner offset
+        descan_offset_cur = self.descanner.scanOffset.value
+
+        # Run image translation pre-align
+        f = fastem.align(self.scanner, self.multibeam, self.descanner, self.mppc, self.stage, self.ccd,
+                         self.beamshift, self.det_rotator, calibrations)
+
+        e = f.result(timeout=900)
+
+        self.assertIsNone(e)  # check no exceptions were returned
+
+        # get the calibrated descanner offset
+        descan_offset_calib = self.descanner.scanOffset.value
+
+        # check the calibrated descan offset is different from the previous offset
+        self.assertNotEqual(descan_offset_cur, descan_offset_calib)
+
+    def test_progress(self):
+        """Check if some progress is reported during the optical autofocus calibration."""
+
+        self.updates = 0  # updated in callback on_progress_update
+
+        calibrations = [OPTICAL_AUTOFOCUS, IMAGE_TRANSLATION_PREALIGN]
+        f = fastem.align(self.scanner, self.multibeam, self.descanner, self.mppc, self.stage, self.ccd,
+                         self.beamshift, self.det_rotator, calibrations)
+
+        f.add_update_callback(self.on_progress_update)  # callback executed every time f.set_progress is called
+        f.add_done_callback(self.on_done)  # callback executed when f.set_result is called (via bindFuture)
+
+        e = f.result()
+
+        self.assertIsNone(e)  # check no exceptions were returned
+        self.assertTrue(self.done)
+        # at least one update per calibration plus once at start of calibration, plus once at end of calibration
+        self.assertGreaterEqual(self.updates, 4)
+
+    def test_cancel(self):
+        """Test if it is possible to cancel the optical autofocus calibration."""
+
+        # FIXME no subfuture available yet, which are cancelable
+        #  when subfutures are implemented, add a check in this test case that the subfuture was also cancelled
+
+        self.end = None  # updated in callback on_progress_update
+        self.updates = 0  # updated in callback on_progress_update
+        self.done = False  # updated in callback on_done
+
+        calibrations = [OPTICAL_AUTOFOCUS]
```

Comment:
Try with more calibrations! Anyway, only the first one should be started anyway.

Reply:
We can only try with 2, as all the others rely on HW. I will add the second one.